pyccl/base/parameters/parameters_base.py

```diff
@@ -74,6 +74,10 @@ def _new_setattr(self, key, value):
 
     def __init__(self):
         # Create a new instance if a factory is provided.
+        if not (hasattr(self, "_instance") or hasattr(self, "_factory")):
+            name = type(self).__name__
+            raise TypeError(f"Can't instantiate {name} with no set "
+                            "`instance` or `factory`.")
```

Comment:
Please explain what you did in this file and why

Reply:
done

Reply:
No, I mean explain this to me. No idea what the comment means

Reply:
prevents `CCLParameters` from getting instantiated